stmhal/system_stm32.c

```diff
@@ -409,8 +420,16 @@ void SystemClock_Config(void)
 
     RCC_ClkInitStruct.AHBCLKDivider = RCC_SYSCLK_DIV1;
     #if defined(MCU_SERIES_F4) || defined(MCU_SERIES_F7)
-    RCC_ClkInitStruct.APB1CLKDivider = RCC_HCLK_DIV4;
-    RCC_ClkInitStruct.APB2CLKDivider = RCC_HCLK_DIV2;
+    
+    /*Set up the peripheral bus clocks*/
+    #if !defined MICROPY_HW_CLK_ABP1DIV
+        #define MICROPY_HW_CLK_ABP1DIV RCC_HCLK_DIV4
+    #endif
+    #if !defined MICROPY_HW_CLK_ABP2DIV
+        #define MICROPY_HW_CLK_ABP2DIV RCC_HCLK_DIV2
+    #endif
```

Comment:
Would be better to put these auto-config lines at the start of the file (starting at line 111).  Then it can work with F4/F7 and L4 by defaulting to DIV4/DIV2 for F4/F7, and defaulting to DIV1/DIV1 for the L4.  Then the 2 lines below can be the same for all the MCUs.